### PR TITLE
Make entity properties be undefined after entity is unloaded

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Instead of having default "zero" values.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by ctrlaltdavid** [comment]
QA should include exercising demo content while checking program log for JavaScript errors - this change could expose some latest JavaScript bugs.

Scripts attached for demoing behaviour with Interface and entity scripts:
- Interface script:
[entityPropertiesAfterDeleteA.zip](https://github.com/highfidelity/hifi/files/102007/entityPropertiesAfterDeleteA.zip)
- entity script:
[entityPropertiesAfterDeleteB.zip](https://github.com/highfidelity/hifi/files/102008/entityPropertiesAfterDeleteB.zip)

**Comment by jherico** [comment]
add to whitelist

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9429f48a7cd2b7a87a49bbb341891f449f6f2617 available at: https://gist.github.com/8ab874a868c34e2dc14a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9429f48a7cd2b7a87a49bbb341891f449f6f2617 available at: https://gist.github.com/8ab874a868c34e2dc14a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9429f48a7cd2b7a87a49bbb341891f449f6f2617 available at: https://gist.github.com/8ab874a868c34e2dc14a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 840bf7cfb4c3d239df05ca8dc6c50fa8a722bd42 available at: https://gist.github.com/8ab874a868c34e2dc14a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.dmg

File: libraries/entities/src/EntityItemProperties.cpp [hunk 1]
```diff
@@ -47,7 +47,9 @@ _localRenderAlphaChanged(false),
 _defaultSettings(true),
 _naturalDimensions(1.0f, 1.0f, 1.0f),
 _naturalPosition(0.0f, 0.0f, 0.0f),
-_desiredProperties(desiredProperties)
+_desiredProperties(desiredProperties),
+
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItemProperties.cpp]: won't block merge - but... in the future:

- where possible, migrate these initializers to the header file using `{ value }` style notation
- why the extra blank line

**Comment by ZappoMan** [comment]
@ctrlaltdavid - do you have an existing entity script that demonstrates the fact that on unload, the property values are undefined? what other repro/tests do you recommend for testing this?

**Comment by ctrlaltdavid** [comment]
@ZappoMan I attached a couple of scripts in comment, above.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8a8e14f54a027a72cb17d45af2c6e67c4650337e available at: https://gist.github.com/8ab874a868c34e2dc14a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6919.dmg

**Comment by OmegaHeron** [comment]
Tested on Windows 10 - compared to released Interface 4038.

On released build - added cube, placed B variant script in cube - ran A variant script - deleted cube.
Saw properties still exist for deleted cube in log.  Repeated with this build log showed "undefined" for cube properties matching behavior @ctrlaltdavid suggests is correct.  This passes for test case outlined.  OS X testing pending - will update.

**Comment by OmegaHeron** [comment]
Can't test on OS X - release build for it has become unusable (as of build 4038) will document that elsewhere.

**Pull request merged**